mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb

```diff
@@ -120,14 +120,16 @@ func TestFrameworkProvider_LoadAndValidateFramework_project(t *testing.T) {
 			ExpectedConfigStructValue: types.StringValue("project-from-GOOGLE_PROJECT"),
 		},
 		// Handling unknown values
-		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14444
-		// "when project is an unknown value, the provider treats it as if it's unset (align to SDK behaviour)": {
-		// 	ConfigValues: fwmodels.ProviderModel{
-		// 		Project: types.StringUnknown(),
-		// 	},
-		// 	ExpectedDataModelValue:    types.StringNull(),
-		// 	ExpectedConfigStructValue: types.StringNull(),
-		// },
+		"when project is an unknown value, the provider treats it as if it's unset and uses an environment variable instead": {
```

Comment:
The actual use case here is that eventually the value will be known and the provider will authenticate with the now-known value, right? Or did this just never work?

Reply:
> eventually the value will be known and the provider will authenticate with the now-known value

Yeah this is right - in the original GH issue around the `credentials` field the SDK code would see the unknown value for credentials as an empty string and treated like it was not provided at all. During the process of applying a plan the provider re-authenticates multiple times, and eventually the value of credentials changes from `""`  to the credentials information. It's like 'just-in-time' credentials.

I imagine it's the same for `project`, but I don't think it's good practice to rely on this behaviour to be honest!

Reply:
That tracks with me! And we're removing the surprising behaviour for both SDK + Framework with `5.0.0`, right?

Reply:
We're removing how empty strings get ignored in 5.0.0 (e.g if a user sets project = "" in their provider block it'll stop the GOOGLE_PROJECT env being used in 5.0+), but there isn't a similar correction we need to make in 5.0 for unknown values, I think.